nncf/tensorflow/quantization/algorithm.py

```diff
@@ -66,8 +70,100 @@
 ]
 
 
+class QuantizationPointStateNames:
+    QUANTIZER_SPEC = 'quantizer_spec'
+    TARGET_POINT = 'target_point'
+    TARGET_POINT_CLASS_NAME = 'target_point_class_name'
+    OP_NAME = 'op_name'
+
+
+class QuantizationPoint:
+    _state_names = QuantizationPointStateNames
+
+    def __init__(self, op_name: str, quantizer_spec: TFQuantizerSpec, target_point: TargetPoint):
+        self.target_point = target_point
+        self.op_name = op_name
+        self.quantizer_spec = quantizer_spec
+
+    def is_weight_quantization(self) -> bool:
+        return isinstance(self.target_point, TFLayerWeight)
+
+    def get_state(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary with Python data structures (dict, list, tuple, str, int, float, True, False, None) that
+        represents state of the object.
+        """
+        return {
+            self._state_names.TARGET_POINT: self.target_point.get_state(),
+            self._state_names.TARGET_POINT_CLASS_NAME: self.target_point.__class__.__name__,
+            self._state_names.QUANTIZER_SPEC: self.quantizer_spec.get_state(),
+            self._state_names.OP_NAME: self.op_name
+        }
+
+    @classmethod
+    def from_state(cls, state: Dict[str, Any]) -> 'QuantizationPoint':
+        """
+        Creates the object from its state.
+
+        :param state: Output of `get_state()` method.
+        """
+        target_point_cls = TF_STATEFUL_CLASSES.get_registered_class(state[cls._state_names.TARGET_POINT_CLASS_NAME])
+        kwargs = {cls._state_names.TARGET_POINT: target_point_cls.from_state(state[cls._state_names.TARGET_POINT]),
+                  cls._state_names.QUANTIZER_SPEC: TFQuantizerSpec.from_state(state[cls._state_names.QUANTIZER_SPEC]),
+                  cls._state_names.OP_NAME: state[cls._state_names.OP_NAME]}
+        return cls(**kwargs)
```

Comment:
Have to introduce registry with stateful classes for creation TargetPoint from the state.
The problem is that `TargetPoint` has a lot of successors (TFLayerPoint, TFLayer, TFBeforeLayer...) and not clear which class to use for de-serialization object from state dict.

`load_state` paradigm works if we already created certain target point, but these classes have a mandatory parameters on init. It's not convenient to make all this parameters optional, create not-consistent dummy objects and then initialize  them by load_state properly. 

Instead, this registry allows to deserialize objects directly without introducing non-consistent objects and removing mandatory parameters on init